src/group_theory/abelianization.lean

```diff
@@ -96,4 +106,24 @@ begin
   refl,
 end
 
+section lift
+
+variables {B : Type v} [comm_group B]
+
+/-- The bijection underlying the abelianization-forgetful adjuction from groups to abelian groups.
+-/
+def hom_equiv :
+(abelianization G →* B) ≃ (G →* B) :=
+{ to_fun := λ f, { to_fun := f.1 ∘ abelianization.of ,
+  map_one' := by simp,
+  map_mul' := by simp } ,
+  inv_fun := λ g, abelianization.lift g,
+  left_inv := by { intro x, ext, simp },
+  right_inv := by { intro x, ext, simp } }
+
+@[simp]
+lemma hom_equiv_apply (f) (x) : (hom_equiv f : G →* B) x = f (of x) := rfl
```

Comment:
I think `@[simps]` before the `def` above will generate this simp lemma and more. (Like `symm_apply`.)

Reply:
Thanks. Done.